core-bundle/src/Resources/contao/modules/ModuleRegistration.php

```diff
@@ -89,6 +89,17 @@ protected function compile()
 			}
 		}
 
+		$strFormId = 'tl_registration_' . $this->id;
+
+		// Purge expired registrations (#3709)
+		if (Input::post('FORM_SUBMIT') == $strFormId && $email = Input::post('email'))
+		{
+			foreach (MemberModel::findExpiredRegistrations(array('column' => array('email = ?'), 'value' => array($email))) ?? array() as $member)
```

Comment:
How can this work? IIRC, `$arrOptions` is ignored in the model:

https://github.com/contao/contao/blob/61cf04641590af805a371cbc7d900f0b6eefe9fa/core-bundle/src/Resources/contao/models/MemberModel.php#L241-L255

Reply:
How about adding a `findExpiredRegistrationsByEmail()` method to the `MemberModel` class? Or maybe we can adjust the existing query if `$arrOptions['email']` is set?

Reply:
I'd vote for the former. The `findExpiredRegistrations` method having an `$arrOptions` argument is likely unintended.